src/main/java/keqing/Keqing.java

```diff
@@ -0,0 +1,204 @@
+        System.out.println("3. deadline: add a task with deadline to the list;");
+        System.out.println("4. event: add a task with specific starting and ending time");
+        System.out.println("5, mark: mark a task as 'done' state;");
+        System.out.println("6. unmark: unmark a task from 'done' state;");
+        System.out.println(LINE);
+    }
+
+    public static void readToDo(Task[] tasks, String content) throws IllegalInputException {
+        if (content.equals("todo")) {
+            throw new IllegalInputException("Keqing doesn't understand what you actually want to do...");
+        }
+        else {
+            ToDo toDoTask = new ToDo(content, Task.getTaskCount());
+            tasks[Task.getTaskCount() - 1] = toDoTask;
+        }
+    }
+
+    public static void readDeadline(Task[] tasks, String content) throws IllegalInputException {
+        if (content.contains("/by")) {
+            int indexOfBy = content.indexOf("/by");
+            if (indexOfBy + 3 < content.length()) {
+                String by = content.substring(indexOfBy + 3).trim();
+                Deadline deadlineTask = new Deadline(content, Task.getTaskCount(), by);
+                tasks[Task.getTaskCount() - 1] = deadlineTask;
+            }
+            else {
+                throw new IllegalInputException("Keqing doesn't think your input makes sense...");
+            }
+        }
+        else {
+            throw new IllegalInputException("Please check if you have typed in a valid deadline.");
+        }
+    }
+
+    public static void readEvent(Task[] tasks, String content) throws IllegalInputException {
+        if (content.contains("./from") && content.contains("./to")) {
+            int indexOfFrom = content.indexOf("/from");
+            int indexOfTo = content.indexOf("/to");
+            if (indexOfFrom < indexOfTo) {
+                String from = content.substring(indexOfFrom + 5, indexOfTo).trim();
+                String to = content.substring(indexOfTo + 3).trim();
+                Event eventTask = new Event(content, Task.getTaskCount(), from, to);
+                tasks[Task.getTaskCount() - 1] = eventTask;
+            }
+            else {
+                throw new IllegalInputException("Keqing doens't think your input makes sense...");
+            }
+        }
+        else {
+            throw new IllegalInputException("Please check if you have typed in the event duration in a valid form...");
+        }
+    }
+
+    public static void doCommand(Task[] tasks, String text) throws IllegalInputException {
+        String splittedText[] = text.split(" ", 2);
+        String command = splittedText[0];
+        String content = splittedText[splittedText.length - 1];
+        switch (command) {
+        case "list":
+            printTaskList(tasks);
+            break;
+        case "menu":
+            printMenu();
+            break;
+        case "mark":
```

Comment:
Missing fallthrough comment. Take a look at java coding standard document on this.